source/patterns/@aws-solutions-constructs/aws-fargate-opensearch/lib/index.ts

```diff
@@ -0,0 +1,210 @@
+   * Optional Name for the container environment variable set to the domain endpoint.
+   *
+   * @default - DOMAIN_ENDPOINT
+   */
+  readonly domainEndpointEnvironmentVariableName?: string;
+}
+
+export class FargateToOpenSearch extends Construct {
+  public readonly vpc: ec2.IVpc;
+  public readonly service: ecs.FargateService;
+  public readonly container: ecs.ContainerDefinition;
+  public readonly userPool: cognito.UserPool;
+  public readonly userPoolClient: cognito.UserPoolClient;
+  public readonly identityPool: cognito.CfnIdentityPool;
+  public readonly openSearchDomain: opensearch.CfnDomain;
+  public readonly openSearchRole: iam.Role;
+  public readonly cloudWatchAlarms?: cloudwatch.Alarm[];
+
+  constructor(scope: Construct, id: string, props: FargateToOpenSearchProps) {
+    super(scope, id);
+    defaults.CheckProps(props);
+    defaults.CheckFargateProps(props);
+
+    this.vpc = defaults.buildVpc(scope, {
+      existingVpc: props.existingVpc,
+      defaultVpcProps: props.publicApi ? defaults.DefaultPublicPrivateVpcProps() : defaults.DefaultIsolatedVpcProps(),
+      userVpcProps: props.vpcProps,
+      constructVpcProps: { enableDnsHostnames: true, enableDnsSupport: true }
+    });
+
+    if (props.existingFargateServiceObject) {
+      this.service = props.existingFargateServiceObject;
+      // CheckFargateProps confirms that the container is provided
+      this.container = props.existingContainerDefinitionObject!;
+    } else {
+      [this.service, this.container] = defaults.CreateFargateService(
+        scope,
+        id,
+        this.vpc,
+        props.clusterProps,
+        props.ecrRepositoryArn,
+        props.ecrImageVersion,
+        props.fargateTaskDefinitionProps,
+        props.containerDefinitionProps,
+        props.fargateServiceProps
+      );
+    }
+
+    let cognitoAuthorizedRole: iam.Role;
+
+    [this.userPool, this.userPoolClient, this.identityPool, cognitoAuthorizedRole] =
+      defaults.buildCognitoForSearchService(this, props.cognitoDomainName ?? props.openSearchDomainName);
+
+    let securityGroupIds;
+
+    if (this.vpc) {
+      securityGroupIds = defaults.getServiceVpcSecurityGroupIds(this.service);
+    }
+
+    const buildOpenSearchProps: defaults.BuildOpenSearchProps = {
+      userpool: this.userPool,
+      identitypool: this.identityPool,
+      cognitoAuthorizedRoleARN: cognitoAuthorizedRole.roleArn,
+      vpc: this.vpc,
+      openSearchDomainName: props.openSearchDomainName,
+      clientDomainProps: props.openSearchDomainProps,
+      securityGroupIds
+    };
+
+    [this.openSearchDomain, this.openSearchRole] = defaults.buildOpenSearch(this, buildOpenSearchProps);
+
+    if (props.createCloudWatchAlarms === undefined || props.createCloudWatchAlarms) {
+      this.cloudWatchAlarms = defaults.buildOpenSearchCWAlarms(this);
+    }
+
+    // Add environment variables
+    const bucketArnEnvironmentVariableName = props.domainEndpointEnvironmentVariableName || 'DOMAIN_ENDPOINT';
```

Comment:
pretty sure bucketArnEnvironmentVariableName is not the variable name you want here.